docs/backend-services/OData.md

```diff
@@ -221,6 +221,30 @@ Navigations within navigations are also supported. For example `columns: [{ id:
 The dataset from the backend is automatically extracted and navigation fields are flattened so the grid can display them and sort/filter just work. The exact property that is used as the dataset depends on the oData version: `d.results` for v2, `results` for v3 and `value` for v4. If needed a custom extractor function can be set through `oDataOptions.datasetExtractor`.
 For example if the backend responds with `{ value: [{ id: 1, nav1: { field1: 'x' }, { nav2: { field2: 'y' } } ] }` this will be flattened to `{ value: [{ id: 1, 'nav1/field1': 'x', 'nav2/field2': 'y' } ] }`.
 
+### Override the filter query
+
+Column filters may have a `Custom` operator, that acts as a placeholder for you to define your own logic. To do so, the easiest way is to provide the `filterQueryOverride` callback int the OdataOptions. This method will be called with `OdataFilterQueryOverrideArgs` to let you decide dynamically on how the filter should be assembled.
+
+E.g. you could listen for a specific column and the active OperatorType.custom in order to switch the filter to a matchesPattern sql like search:
+
+```ts
+backendServiceApi: {
+  options: {
+    filterQueryOverride: (args) => {
+      const { fieldName, columnDef, columnFilterOperator, searchValue } = args;
+      if (columnFilterOperator === OperatorType.custom && columnDef?.id === 'name') {
+        let matchesSearch = (searchValue as string).replace(/\*/g, '.*');
+        matchesSearch = matchesSearch.slice(0, 1) + '%5E' + matchesSearch.slice(1);
```

Comment:
what is the escape of `%5E`? Maybe add a comment mentioning what and why we're escaping?